### PR TITLE
Fleet Performance Improvements

### PR DESCRIPTION
**Pull request by richard-cox**
### Summary
Contributes to #12424

### Checklist
- [ ] The PR is linked to an issue and the linked issue has a Milestone, or no issue is needed
- [ ] The PR has a Milestone 
- [ ] The PR template has been filled out
- [ ] The PR has been self reviewed 
- [ ] The PR has a reviewer assigned
- [ ] The PR has automated tests or clear instructions for manual tests and the linked issue has appropriate QA labels, or tests are not needed
- [ ] The PR has reviewed with UX and tested in light and dark mode, or there are no UX changes

### CODE REVIEW

File: shell/models/fleet.cattle.io.gitrepo.js [hunk 1]
```diff
@@ -339,9 +344,10 @@ export default class GitRepo extends SteveModel {
           namespacedName = `${ r.namespace }:${ r.name }`;
         }
 
-        let state = r.state;
+        let state = r.state; // FIXME: why assign r.state when it's always overwritten?
+
         const perEntry = r.perClusterState?.find((x) => x.clusterId === c.id);
-        const tooMany = r.perClusterState?.length >= 10 || false;
+        const tooMany = r.perClusterState?.length >= 10 || false; // FIXME: shouldn't we not calc perEntry if there are too many?
```
Comment (manno) [inline shell/models/fleet.cattle.io.gitrepo.js:350]: Additionally, Fleet creates an incomplete flag on the  resource and shouldn't add more than 10 perCluster states.

File: shell/models/fleet.cattle.io.gitrepo.js [hunk 2]
```diff
@@ -323,14 +324,18 @@ export default class GitRepo extends SteveModel {
   get resourcesStatuses() {
     const clusters = this.targetClusters || [];
     const resources = this.status?.resources || [];
-    const conditions = this.status?.conditions || [];
+    // const conditions = this.status?.conditions || [];
+    // const bundleDeployments = this.bundleDeployments || [];
 
     const out = [];
 
     for (const c of clusters) {
```
Comment (manno) [inline shell/models/fleet.cattle.io.gitrepo.js:332]: This code assumes each resource is installed on every cluster. This is not always the case. Users can for example use targetCustomization to deploy resources only to some clusters. Additionally, popular charts install different resources depending on the k8s version of the cluster. Some charts even allow to deploy "extraObjects" via a helm chart value. 🤷

File: shell/models/fleet.cattle.io.gitrepo.js
[the remark this answers is not in the document]
Comment (richard-cox) [reply shell/models/fleet.cattle.io.gitrepo.js:1]: ouch, as in the gitrepo resource has a native `targetClusters` property?

**Comment by richard-cox** [comment]
superseded by https://github.com/rancher/dashboard/pull/12521

**Pull request closed without merging**